### PR TITLE
proposed JsPrettier formatting

### PR DESCRIPTION
**Pull request by paOol**
Highly recommend implementing JsPrettier to handle formatting and still maintain ESLinting rules. JSprettier's rules can be tweaked if there are opinionated differences (white spaces and what not). 

Formatting on save will maintain consistency throughout the codebase and linting will be unaffected.

### CODE REVIEW

**Comment by paOol** [comment]
more info:

https://blog.echobind.com/integrating-prettier-eslint-airbnb-style-guide-in-vscode-47f07b5d7d6a

https://prettier.io/docs/en/eslint.html#turn-off-eslint-s-formatting-rules

**Comment by ClemensLey** [comment]
I think integrating JsPrettier is a very good idea. I will wait to give @slashrsm and @lotusbaba a chance to comment before I merge.

**Comment by slashrsm** [comment]
I really like the idea, thank you for suggesting it! The more we can automate the quality enforcement the better. I would have 3 comments: 

1. It seems that JsPrettier becomes really useful once one enables automatic formatting in the IDE (I needed to add `"editor.formatOnSave": true` in VSCode). Would be nice to document this somewhere. 
2. Please use more verbose commit messages. Imagine someone looking at the commit history in a year time and seeing the "jsprettier" and "sample diff" commits? Will the message tell them what they are about? While the first one might give some clue about it the second won't at all. 
3. Plese use rebasing instead of merging when updating the branch. It produces much nicer commit history and clearer pull requests diff. There are a bunch of commits from master appearing in this pull request and its diff at the moment, which makes it hard to determine what is the part of this change and what is noise. Using rebase will solve that.

File: src/publickey.js [hunk 1]
```diff
@@ -365,8 +365,10 @@ PublicKey.prototype._getID = function _getID() {
  * @param {String|Network=} network - Which network should the address be for
  * @returns {Address} An address generated from the public key
  */
-PublicKey.prototype.toAddress = function(network) {
-  var Address = require('./address');
+PublicKey.prototype.toAddress = function (network) {
+  // #weirdstuff resolve circular dependency.
```
Comment (ClemensLey) [inline src/publickey.js]: I think we should leave this comment here because we wanted to come back to this later to resolve

**Comment by ClemensLey** [comment]
Just looked at the changed files. Everything looks good to me, great first contribution. I have just one question: You removed 'use strict' from the files. Could you explain why this is not necessary anymore?

**Comment by paOol** [comment]
Hi,

since this was my first PR before seeing the contrib guide, it was not rebased. Thus my changes were merged with slashrsm. My changes did not include the 'use strict' removal, but my changes were bundled with slashrsm. 

I'll be creating new branches and rebasing from here on.

**Comment by lotusbaba** [comment]
Just curious -- Commit - 618169a, fef7bf1 & ed5a18e, e81c11f are same and a28f3de is empty.

Any reason this is so?

**Pull request merged**